### ISSUE TITLE
[BUG] webkit timeout on ubuntu jammy based image

### ISSUE THREAD

**Issue by thebuchanan3** [microsoft/playwright#15764]
### Workaround

Use **Ubuntu 22** to run the **Ubuntu 22** (jammy) container, then it works.
Using Ubuntu 20 to run it, won't work and leads into this issue.
Please upvote if you encounter this issue, this helps us to prioritise. 

---

Root issue: different set of kernel syscalls. See https://stackoverflow.com/questions/32841982/how-can-docker-run-distros-with-different-kernels/56606244#56606244

---

**Context:**
- Playwright Version: 1.23.0
- Operating System: Linux
- Python version: 3.10
- Browser: Webkit

**Code Snippet**
pw.py
```python
import asyncio
from playwright.async_api import async_playwright


async def run():
    async with async_playwright() as playwright:
        await playwright.webkit.launch(headless=True)

if __name__ == "__main__":
    asyncio.run(run())
```

Dockerfile
```dockerfile
FROM snakepacker/python:3.10

RUN apt-install python3-pip && \
    pip install playwright==1.23.0 && \
    playwright install --with-deps webkit

ADD ./pw.py pw.py
CMD python3 pw.py
```

**Describe the bug**

playwright times out starting webkit

```sh
docker build -t test-pw . && docker run test-pw
```

```
Traceback (most recent call last):
  File "//pw.py", line 11, in <module>
    asyncio.run(run())
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "//pw.py", line 8, in run
    await playwright.webkit.launch(headless=True)
  File "/usr/local/lib/python3.10/dist-packages/playwright/async_api/_generated.py", line 11399, in launch
    await self._impl_obj.launch(
  File "/usr/local/lib/python3.10/dist-packages/playwright/_impl/_browser_type.py", line 93, in launch
    Browser, from_channel(await self._channel.send("launch", params))
  File "/usr/local/lib/python3.10/dist-packages/playwright/_impl/_connection.py", line 43, in send
    return await self._connection.wrap_api_call(
  File "/usr/local/lib/python3.10/dist-packages/playwright/_impl/_connection.py", line 370, in _
    return await result
  File "/usr/local/lib/python3.10/dist-packages/playwright/_impl/_connection.py", line 78, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.TimeoutError: Timeout 30000ms exceeded.
=========================== logs ===========================
<launching> /root/.cache/ms-playwright/webkit-1668/pw_run.sh --inspector-pipe --headless --no-startup-window
<launched> pid=440
============================================================
```

**Comment by rwoll** [comment]
Moving to main repo per @aslushnikov's request. This seems to be a non-Python-specific issue, and more with install script. The base image is on a technically supported OS.

/cc @yury-s

**Comment by aslushnikov** [comment]
@thebuchanan3 what's your host operating system that runs this docker? Do you run something like Ubuntu 18 Bionic on a host and Ubuntu 20.04 Focal container?

**Comment by thebuchanan3** [comment]
> @thebuchanan3 what's your host operating system that runs this docker? Do you run something like Ubuntu 18 Bionic on a host and Ubuntu 20.04 Focal container?

i ran a jammy container on a focal host

**Comment by aslushnikov** [comment]
@thebuchanan3 so there are 2 work-arounds:

- Use Ubuntu 22 to run the Ubuntu 22 (jammy) container, then it works.
- Run container with `--privileged` flag

This is due to docker architecture and kernel re-use: it looks like WebKit for Jammy is using something that kernel from focal can't provide in unprivileged mode.

**Comment by mirao** [comment]
I can confirm that WebKit freezes if you start it in a docker container (guest) based on Ubuntu 22.04 on a host with Ubuntu 20.04.
It works well on guest Ubuntu 22.04 running on host Ubuntu 22.04.
I didn't try the workaround `--privileged` (isn't it a bit dangerous?). I'm staying on a docker image based on Ubuntu 20.04 instead because I need to run it on different host distros (Ubuntu 20.04 and 22.04).

**Comment by aslushnikov** [comment]
So this is a docker-specific issue rather then a WebKit-specific: containers are only abstracting user namespace, whereas
kernel is shared with host. See https://stackoverflow.com/questions/32841982/how-can-docker-run-distros-with-different-kernels/56606244#56606244

It looks like Webkit for Ubuntu 22 requires some syscall from Ubuntu 22.04 kernel that is not available in Kernel from Ubuntu 20.04, so thus the bug. 

I'll close this for now since there's no action item for the Playwright team here.

**Comment by mxschmitt** [comment]
I was looking a bit more into it, the issue is that the following subprocess from WebKit hangs:

`/usr/lib/x86_64-linux-gnu/gstreamer1.0/gstreamer-1.0/gst-plugin-scanner -l /ms-playwright/webkit-1808/minibrowser-wpe/bin/MiniBrowser`

probably due to https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/937.

To workaround it, you can just `ln -s /usr/bin/echo /usr/lib/x86_64-linux-gnu/gstreamer1.0/gstreamer-1.0/gst-plugin-scanner` inside the Docker contrainer.

You can also try to upgrade glib on the Docker host, since it [seems fixed](https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2435) inside glib already.

**Comment by mxschmitt** [comment]
Heads up: I can confirm that this will be fixed as part of 1.34.
